### ISSUE TITLE
I could get accuracy 0% as results 

### ISSUE THREAD

**Issue by minkyukang707** [wimh966/QDrop#11]
I'm trying to get a similar results in github but I could only get abnormal results, accuracy : 0% 

I used ImageNet datasets, resnet18 architecture 

I'll attach some pictures:

- ImageNet folder, directory structure 
- the code that I executed 
- architecture and results screent shots

Thank you so much 

train directory
- ILSVRC/Data/CLS-LOC/train/folders/images.jpeg
![train](https://user-images.githubusercontent.com/87360858/197084903-4643857a-d106-427c-868d-c7fee01df8c7.PNG)

val directory
-ILSVRC/Data/CLS-LOC/val2/val(single folder)/images.jpeg
![val9](https://user-images.githubusercontent.com/87360858/197084911-36b7f664-bc22-4c39-99d4-08238093c677.PNG)

code that I executed 
![run10](https://user-images.githubusercontent.com/87360858/197085036-f924fafe-2060-4c20-bdca-665d74f0fefb.PNG)

Architecture & results 
![1](https://user-images.githubusercontent.com/87360858/197085075-9759d80e-977b-4c6b-9bc3-8a0bc1510b88.PNG)

![2](https://user-images.githubusercontent.com/87360858/197085103-37d1e7f2-637e-40d9-b3ea-29ffe68b2313.PNG)

![3](https://user-images.githubusercontent.com/87360858/197085111-b0f963a1-733c-4330-980a-0c9d52667a2d.PNG)

**Comment by wimh966** [comment]
It seems that you are trying to achieve the results of Case 3 in paper? If so, there should be something wrong. Before that, have you tried the full-precision results and also the results of QDrop.

**Comment by kirIn0305** [comment]
Thank you for sharing your great job!

I'm trying to get similar results in GitHub too.
But I got almost the same result as @minkyukang707 .

## Execution Procedure

### 1. a modification for hubconf.py

Your pre-trained models in hubconf.py are located in your local PC directory.
Therefore, I replace your hubconf.py with BRECQ hubconf.py.
(https://github.com/yhhhli/BRECQ/blob/main/hubconf.py)
Then, I verified the accuracy of FP32.
I got the expected accuracies of them.
(I checked the result by inserted  ``print('FP32 model accuracy: {}'.format(validate_model(test_loader, cnn)))`` in main_imagenet.py:L198.)

Here is the result of ResNet18.

```bash
Test: [  0/782] Time  0.618 ( 0.618)    Acc@1  85.94 ( 85.94)   Acc@5  95.31 ( 95.31)
Test: [100/782] Time  0.283 ( 0.098)    Acc@1  85.94 ( 78.12)   Acc@5  98.44 ( 93.05)
Test: [200/782] Time  0.360 ( 0.093)    Acc@1  81.25 ( 78.09)   Acc@5  95.31 ( 94.15)
Test: [300/782] Time  0.392 ( 0.095)    Acc@1  76.56 ( 77.98)   Acc@5  98.44 ( 94.20)
Test: [400/782] Time  0.008 ( 0.093)    Acc@1  68.75 ( 74.98)   Acc@5  95.31 ( 92.56)
Test: [500/782] Time  0.214 ( 0.092)    Acc@1  79.69 ( 73.38)   Acc@5  93.75 ( 91.34)
Test: [600/782] Time  0.307 ( 0.092)    Acc@1  78.12 ( 72.16)   Acc@5  90.62 ( 90.60)
Test: [700/782] Time  0.295 ( 0.091)    Acc@1  70.31 ( 71.08)   Acc@5  90.62 ( 89.96)
 * Acc@1 71.004 Acc@5 89.964
 FP32 model accuracy: 71.00399780273438
 ```
 
 ### 2. Run scripts generated by exp/eonfig.sh
 
 I ran exp/w2a2/resnet18/run.sh. Then, I got the below result.
 
 ```bash
 Test: [  0/782] Time  0.676 ( 0.676)    Acc@1   0.00 (  0.00)   Acc@5   0.00 (  0.00)
Test: [100/782] Time  0.311 ( 0.102)    Acc@1   0.00 (  0.29)   Acc@5   0.00 (  1.10)
Test: [200/782] Time  0.310 ( 0.096)    Acc@1   0.00 (  0.15)   Acc@5   0.00 (  0.65)
Test: [300/782] Time  0.407 ( 0.098)    Acc@1   0.00 (  0.12)   Acc@5   0.00 (  0.57)
Test: [400/782] Time  0.221 ( 0.096)    Acc@1   0.00 (  0.09)   Acc@5   0.00 (  0.49)
Test: [500/782] Time  0.326 ( 0.095)    Acc@1   0.00 (  0.25)   Acc@5   7.81 (  1.04)
Test: [600/782] Time  0.291 ( 0.094)    Acc@1   0.00 (  0.31)   Acc@5   0.00 (  1.15)
Test: [700/782] Time  0.239 ( 0.093)    Acc@1   0.00 (  0.31)   Acc@5   0.00 (  1.26)
 * Acc@1 0.312 Acc@5 1.244
Full quantization (W2A2) accuracy: 0.31200000643730164
```

I wonder if I'm doing this incorrectly!?

Thank you!

**Comment by wimh966** [comment]
The result of yours is not right. minkyukang707 does not run the same setting with you. let's just focus on your result. Can you show me results of W4A4 (you only need to change the bit of weight and activation to 4-bit ) so that I can help you get the right result?

**Comment by kirIn0305** [comment]
I ran exp/w4a4/run.sh
Here is the detail of args.
```bash
#!/bin/bash
PYTHONPATH=../../../..:$PYTHONPATH; \
data_path=/media/ssd1/datasets/public/ILSVRC2012/; \
python ../../../main_imagenet.py --data_path $data_path \
--arch resnet18 --n_bits_w 4 --channel_wise --n_bits_a 4  --act_quant --order together --wwq --waq --awq --aaq \
--weight 0.01 --input_prob 0.5 --prob 0.5
```

I got almost the same as w2a2.

I attached the stdout and stderr.
[w4a4_resnet18.log](https://github.com/wimh966/QDrop/files/10085874/w4a4_resnet18.log)

```bash
$ cd exp/w4a4/resnet18
$ ./run.sh &> w4a4_resnet18.log
~~~
Test: [  0/782]	Time  0.686 ( 0.686)	Acc@1   0.00 (  0.00)	Acc@5   0.00 (  0.00)
Test: [100/782]	Time  0.345 ( 0.105)	Acc@1   0.00 (  0.56)	Acc@5   0.00 (  2.00)
Test: [200/782]	Time  0.307 ( 0.099)	Acc@1   0.00 (  0.28)	Acc@5   0.00 (  1.22)
Test: [300/782]	Time  0.397 ( 0.099)	Acc@1   0.00 (  0.21)	Acc@5   0.00 (  1.02)
Test: [400/782]	Time  0.148 ( 0.096)	Acc@1   0.00 (  0.19)	Acc@5   0.00 (  1.06)
Test: [500/782]	Time  0.158 ( 0.095)	Acc@1   0.00 (  0.48)	Acc@5  15.62 (  1.83)
Test: [600/782]	Time  0.112 ( 0.094)	Acc@1   0.00 (  0.53)	Acc@5   0.00 (  1.97)
Test: [700/782]	Time  0.014 ( 0.093)	Acc@1   0.00 (  0.51)	Acc@5   0.00 (  2.19)
 * Acc@1 0.542 Acc@5 2.256
Full quantization (W4A4) accuracy: 0.5419999957084656
```

**Comment by wimh966** [comment]
In the log, I only saw layer reconstruction rather than block reconstruction? If block reconstruction, it will print "reconstruction xxx for block xxx". So, there are some problems. You can have a check on your operations. Also, you can try another branch "qdrop" (git checkout qdrop). It is easier to use.

**Comment by kirIn0305** [comment]
I changed the the qdrop branch.
I can get the results like your paper.
I used the below fp32 resnet18.
https://github.com/yhhhli/BRECQ/releases/download/v1.0/resnet18_imagenet.pth.tar

I got almost the same results as yours.
Thank you for your kind help!

|Methods | Bits (W/A) | Res18 |
|:---          | :---            | :---     | 
|Full Prec. | 32/32 | 71.004 |
|QDrop | 4/4 | 69.112 |
|QDrop | 2/4 | 65.552 |
|QDrop | 3/3 | 64.508 |
|QDrop | 2/2 | 51.576 |


BTW, @minkyukang707 may want to evaluate case 3 in the paper.

In case 3, all the activations in the target block reconstruction layers are fp32.
Is my understanding correct?
If true, Can I run case 3 block reconstruction using the qdrop branch?

**Comment by wimh966** [comment]
Sorry for the late reply. 
Yes, that's right. 
You can use the qdrop branch to run Case 3, but need to temporarily close the activation quantization of the current block reconstruction.
